admin/javascript/src/components/form-action/index.js

```diff
@@ -0,0 +1,119 @@
+import React from 'react';
+import SilverStripeComponent from 'silverstripe-component';
+
+class FormActionComponent extends SilverStripeComponent {
+  constructor(props) {
+    super(props);
+
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  render() {
+    const props = {
+      type: this.props.type,
+      className: this.getButtonClasses(),
+      disabled: this.props.disabled,
+    };
+
+    if (typeof this.props.id !== 'undefined') {
+      props.id = this.props.id;
+    }
+
+    return (
+      <button {...props}>
+        {this.getLoadingIcon()}
+        {this.props.label}
+      </button>
+    );
+  }
+
+  /**
+   * Returns the necessary button classes based on the given props
+   *
+   * @returns string
+   */
+  getButtonClasses() {
+    const buttonClasses = ['btn'];
+
+    // Add 'type' class
+    buttonClasses.push(`btn-${this.props.style}`);
+
+    // If there is no text
+    if (typeof this.props.label === 'undefined') {
```

Comment:
Makes less sense for the one just below though (font-icon-0 I can just about see). If this is standard idiom for React, fine, it's just a lot slower, and does odd things if set to unexpected falsey values (font-icon-null).